turbomole/default.nix

```
@@ -0,0 +1,147 @@
+  };
+
+  postPatch = ''
+    for f in Config_turbo_env Config_turbo_env.csh scripts mpirun_scripts mpirun_scripts/IMPI/intel64/lib; do
+      patchShebangs $f
+    done
+  '';
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  /*
+  Sets environment variables, that turbomole potentially uses.
+  */
+  TURBOMOLE_SYSNAME = systemName;
+  PARA_ARCH = if useMPI then "MPI" else "SMP";
+
+  /*
+  Turbomole is too tightly coupled to its directory structure and i cannot really break it.
+  Therefore try to clean as good as possible and put most stuff in share and symlink then to entry
+  points.
+  This roughly follows the ideas in Config_turbo_env
+  */
+  installPhase = ''
+    runHook preInstall
+
+
+    # Copy the entire installation to share/turbomole
+    export TURBODIR=$out/share/turbomole
+    mkdir -p $TURBODIR
+    cp -r * $TURBODIR/.
+
+    # Find all executables which are entry points.
+    exesBin=$(find $TURBODIR/bin/${TURBOMOLE_SYSNAME}_${lib.strings.toLower PARA_ARCH} -type l,f -executable)
+    exesScript=$(find $TURBODIR/scripts/ -type f -executable)
+    exes="$exesBin $exesScript"
+
+    # Wrap up executables and link them to the bin directory.
+    mkdir -p $out/bin
+    for exe in $exes ; do
+      ln -s $exe $out/bin/.
+    done
+
+    # Libraries should be available in the library directory
+    mkdir -p $out/lib
+    for i in $(find $TURBODIR/libso -type f); do
+      ln -s $i $out/lib/.
+    done
+
+    runHook postInstall
+  '';
+
+  dontAutoPatchelf = true;
+
+  libSearchPath = lib.strings.makeSearchPath "lib" [
+    "$out/lib"
+    "$out/share/turbomole/libso/${systemName}_mpi"
+    gcc-unwrapped.lib
+    zlib
+    glibc
+  ];
+
+  postFixup = ''
+    # Patch elf dynamic library paths.
+    for dir in $TURBODIR/libso $TURBODIR/bin/*; do
+      files=$(find $dir -type f -executable -maxdepth 1 ! -name rimp2 ! -name ccsdf12)
+      for f in $files; do
+        # echo "Patching $f now!"
+        patchelf \
+          --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+          --set-rpath "$libSearchPath" \
+          $f
+      done
+    done
+
+    exesToWrap="$(find $TURBODIR/bin -type f -executable) $(find $TURBODIR/scripts -type f -executable -maxdepth 1) $(find $TURBODIR/smprun_scripts -type f -executable ! -name "*.so*") $(find $TURBODIR/mpirun_scripts -type f -executable ! -name "*.so*")"
+    for exe in $exesToWrap; do
+      echo "Wrapping exe: $exe"
+      wrapProgram $exe \
+        --prefix PATH : "${which}/bin" \
+        --prefix PATH : "${coreutils}/bin" \
+        --prefix PATH : "${less}/bin" \
+        --prefix PATH : "${more}/bin" \
+        --set TURBODIR "$TURBODIR" \
+        --set PARA_ARCH "${PARA_ARCH}" \
+        --set TURBOARCH ${systemName}
+    done;
+  '';
+
+  meta = with lib; {
+    description = "General purpose quantum chemistry program. Tools, not Toys!";
+    homepage = "https://www.turbomole.org/";
+    license = licenses.unfree;
+    platforms = platforms.linux;
```

```suggestion
    platforms = [ "x86_64-linux" ];
```
This makes the `assert` in the beginning unnecessary.